### PR TITLE
[Review-ready][DNM] /obj/machinery/atmospherics/components (#10284)

### PR DESCRIPTION
**Pull request by duncathan**
Something went wrong with pushing to #10284; this is the same thing.
This is now ready for review; all that's left for me is to fix merge conflicts.
Any maintainers that wanna give this a look over, please do; @Aranclanos in particular I want to look at this before it's merged.

This PR, at its core, changes the way obj/machinery/atmospherics is organized. Previously, it was split up into atmospherics/pipe, atmospherics/unary, atmospherics/binary, and atmospherics/trinary. It is now split into atmospherics/pipe and atmospherics/components. atmospherics/components is then split into components/unary, components/binary, and components/trinary. Component-specific procs and vars were moved to components.

After this initial split, the rest of the PR went into greatly reducing copy-paste and other undesirable things in components code. Notably, the majority of the unary_base.dm, binary_atmos_base.dm and trinary_base.dm files have been cut down entirely by putting their procs in the new components_base.dm. To accomplish this, components have had their various node, air and parent vars condensed into a list. Now instead of node1, node2, and node3, trinary devices have a list of length three, nodes. This allows us to use loops to make our component procs OOP.

Other changes:
- cleans up oldcode in valve.dm
- adds defines to vent pressure checks
- adds a helper proc update_parents() that sets the update var of each parent of a component to 1
- moves a couple things around (cryo.dm and freezer.dm are unary devices goddammit)

Unless I missed something else, that's the PR in a nutshell.

### CODE REVIEW

**Comment by duncathan** [comment]
I fucked up that last commit

**Comment by duncathan** [comment]
Aran gave the all-clear. This can be merged once Travis finishes.

**Comment by phil235** [comment]
I think this proc atmos...

File: code/ATMOSPHERICS/components/unary_devices/unary_base.dm [hunk 1]
```diff
@@ -51,86 +22,26 @@ Iconnery
...
-	..()
-	parent.update = 1
+/obj/machinery/atmospherics/components/unary/atmosinit()
+	var/list/node_connects = new/list()
+	node_connects.Add(dir)
+	..(node_connects)
 
-/obj/machinery/atmospherics/unary/default_change_direction_wrench(mob/user, obj/item/weapon/wrench/W)
+/obj/machinery/atmospherics/components/unary/default_change_direction_wrench(mob/user, obj/item/weapon/wrench/W)
 	if(..())
```
Comment (duncathan) [inline code/ATMOSPHERICS/components/unary_devices/unary_base.dm]: There's the issue. I forgot  to make this if(!..()).

File: code/ATMOSPHERICS/components/unary_devices/cryo.dm [hunk 2]
```diff
@@ -135,15 +136,17 @@
   *
   * @return nothing
   */
-/obj/machinery/atmospherics/unary/cryo_cell/ui_interact(mob/user, ui_key = "main", datum/nanoui/ui = null)
+/obj/machinery/atmospherics/components/unary/cry_cell/ui_interact(mob/user, ui_key = "main", datum/nanoui/ui = null)
```
Comment (duncathan) [inline code/ATMOSPHERICS/components/unary_devices/cryo.dm]: this looks like the source of the cryo issue

File: code/ATMOSPHERICS/components/unary_devices/Freezer.dm [hunk 3]
```diff
@@ -57,18 +57,16 @@
...
-	return interact(user)
+/obj/machinery/atmospherics/components/unary/cold_sink/freezer/attack_ai(mob/user)	return interact(user)
 
-/obj/machinery/atmospherics/unary/cold_sink/freezer/attack_paw(mob/user)
-	return interact(user)
+/obj/machinery/atmospherics/components/unary/cold_sink/freezer/attack_paw(mob/user)	return interact(user)
 
-/obj/machinery/atmospherics/unary/cold_sink/freezer/attack_hand(mob/user)
-	return interact(user)
+/obj/machinery/atmospherics/components/unary/cold_sink/freezer/attack_paw(mob/user)	return interact(user)
```
Comment (duncathan) [inline code/ATMOSPHERICS/components/unary_devices/Freezer.dm]: haha woops

**Comment by Anonus** [comment]
good job aran

**Comment by Stickymayhem** [comment]
Yeah thanks

**Pull request merged**